### PR TITLE
feat(sdk): Allow custom KAO array templates

### PR DESCRIPTION
**Pull request by dmihalcik-virtru**
This will allow opening of files with multi-kas splits.

Step 1 of #314

While I'm here I also:

1. Lets platform-xtest job run in parallel with other e2e tests
2. Updates config for platform-roundtrip job, since the `provision fixture keycloak` needs a new config file
3. Adds a new phony make target to simplify development, you can now `make cli` to get the cli without having to guess the version number
4. Adds a `prettier ignore` directive so `make format` no longer formats the generated protocol buffer code
5. Updates the fetching of KAS public keys to happen during encrypt, not during client creation. This makes more sense now that we don't know at client start time which kases will actually be involved in the encrypt step

### CODE REVIEW

**Review by pflynn-virtru — DISMISSED** [review]

**Review by pflynn-virtru — DISMISSED** [review]

File: lib/tdf3/src/client/index.ts [hunk 1]
```diff
@@ -384,14 +384,23 @@ export class Client {
...
-        url: kasPublicKey.url,
-        kid: kasPublicKey.kid,
-        publicKey: kasPublicKey.publicKey,
-        metadata,
+
+    const splits: SplitStep[] = splitPlan || [{ kas: this.kasEndpoint }];
+    encryptionInformation.keyAccess = await Promise.all(
+      splits.map(async ({ kas, sid }) => {
+        if (!(kas in this.kasKeys)) {
+          this.kasKeys[kas] = fetchKasPublicKey(kas);
```
Comment (elizabethhealy) [inline lib/tdf3/src/client/index.ts:392]: does the cache have a time limit similar to the go one?

**Review by elizabethhealy — APPROVED** [review]

File: lib/tdf3/src/client/index.ts (hunk 1 above)
Comment (dmihalcik-virtru) [reply lib/tdf3/src/client/index.ts:392]: no it probably should?

Comment (dmihalcik-virtru) [reply lib/tdf3/src/client/index.ts:392]: however this is an existing feature; I'm just changing it to be lazily loaded while before it was loaded during the client constructor

Comment (dmihalcik-virtru) [reply lib/tdf3/src/client/index.ts:392]: https://github.com/opentdf/client-web/issues/326

**Pull request merged**